### PR TITLE
Handle preexisting disable rule comments

### PR DESCRIPTION
**Pull request by asingh04**
Referencing the issue [#1248](https://github.com/microsoft/vscode-eslint/issues/1248)

### CODE REVIEW

**Comment by asingh04** [comment]
@dbaeumer can you have a look into it?

**Comment by dbaeumer** [comment]
Will have a look this month.

File: server/src/eslintServer.ts [hunk 1]
```diff
@@ -1819,16 +1819,30 @@ messageQueue.registerRequest(CodeActionRequest.type, (params) => {
 	}
 
 	function createDisableLineTextEdit(editInfo: Problem, indentationText: string): TextEdit {
+		// if the concerned line is not the first  line of the file
+		if ( editInfo.line - 1 > 0) {
+
+			// check previous line if there is a eslint-disable-next-line comment already present
+			const prevLine = textDocument?.getText(Range.create(Position.create(editInfo.line - 2, 0), Position.create(editInfo.line - 2, Number.MAX_VALUE)));
```
Comment (dbaeumer) [inline server/src/eslintServer.ts:1826]: We should use uinteger.MAX_VALUE which is defined in the latest LSP protocol and used in ESLint.

File: server/src/eslintServer.ts [hunk 2]
```diff
@@ -1819,16 +1819,30 @@ messageQueue.registerRequest(CodeActionRequest.type, (params) => {
...
+			}
+
+		}
 		return TextEdit.insert(Position.create(editInfo.line - 1, 0), `${indentationText}// eslint-disable-next-line ${editInfo.ruleId}${EOL}`);
 	}
 
 	function createDisableSameLineTextEdit(editInfo: Problem): TextEdit {
-		// Todo@dbaeumer Use uinteger.MAX_VALUE instead.
-		return TextEdit.insert(Position.create(editInfo.line - 1, 2147483647), ` // eslint-disable-line ${editInfo.ruleId}`);
+		const currentLine = textDocument?.getText(Range.create(Position.create(editInfo.line - 1, 0), Position.create(editInfo.line -1, Number.MAX_VALUE)));
```
Comment (dbaeumer) [inline server/src/eslintServer.ts:1837]: Same here

File: server/src/eslintServer.ts [hunk 3]
```diff
@@ -1933,8 +1947,7 @@ messageQueue.registerRequest(CodeActionRequest.type, (params) => {
 				if (settings.codeAction.disableRuleComment.location === 'sameLine') {
 					workspaceChange.getTextEditChange({ uri, version: documentVersion }).add(createDisableSameLineTextEdit(editInfo));
 				} else {
-					// Todo@dbaeumer Use uinteger.MAX_VALUE instead.
-					const lineText = textDocument.getText(Range.create(Position.create(editInfo.line - 1, 0), Position.create(editInfo.line - 1, 2147483647)));
+					const lineText = textDocument.getText(Range.create(Position.create(editInfo.line - 1, 0), Position.create(editInfo.line - 1, Number.MAX_VALUE)));
```
Comment (dbaeumer) [inline server/src/eslintServer.ts:1950]: Same here

**Review by dbaeumer — CHANGES_REQUESTED** [review]
Can you please exclude the package-lock changes from the PR. Otherwise very hard to merge.

File: server/src/eslintServer.ts (hunk 1 above)
Comment (asingh04) [reply server/src/eslintServer.ts]: changed to `uinteger.MAX_VALUE`

File: server/src/eslintServer.ts (hunk 2 above)
Comment (asingh04) [reply server/src/eslintServer.ts]: changed to `uinteger.MAX_VALUE`

File: server/src/eslintServer.ts (hunk 3 above)
Comment (asingh04) [reply server/src/eslintServer.ts]: changed to `uinteger.MAX_VALUE`

**Comment by asingh04** [comment]
> Can you please exclude the package-lock changes from the PR. Otherwise very hard to merge.

Reverted the change in all `package-lock.json` files

**Comment by asingh04** [comment]
Hi @dbaeumer, the changes you had recommended are done. Please go through this PR.

**Comment by asingh04** [comment]
@dbaeumer resolved the merge conflicts

**Comment by dbaeumer** [comment]
@asingh04 sorry slipped through the cracks. When reviewing I noticed that the code has a lot of `?` and `!` which shouldn't be there. I fixed this. Could you merge again and use the `textDocument` passed in as a param. Makes the code clearer and easier to understand.

Thanks

**Comment by asingh04** [comment]
> @asingh04 sorry slipped through the cracks. When reviewing I noticed that the code has a lot of `?` and `!` which shouldn't be there. I fixed this. Could you merge again and use the `textDocument` passed in as a param. Makes the code clearer and easier to understand.
> 
> Thanks

Oh yes, it is my mistake for not noting that `textDocument` is used with optional chaining in the function, and straight ahead went to use `!` for reading `languageId`. I have made the changes according to you recommendation.

**Comment by dbaeumer** [comment]
Thanks for doing the changes.

**Pull request merged**